dead_pool_reaper.go

```diff
@@ -0,0 +1,168 @@
+package work
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+var (
+	DeadTime   = 5 * time.Minute
+	ReapPeriod = 10 * time.Minute
+)
+
+type deadPoolReaper struct {
+	namespace string
+
+	pool *redis.Pool
+
+	stopChan         chan struct{}
+	doneStoppingChan chan struct{}
+}
+
+func newDeadPoolReaper(namespace string, pool *redis.Pool) *deadPoolReaper {
+	return &deadPoolReaper{
+		namespace:        namespace,
+		pool:             pool,
+		stopChan:         make(chan struct{}),
+		doneStoppingChan: make(chan struct{}),
+	}
+}
+
+func (r *deadPoolReaper) start() {
+	go r.loop()
+}
+
+func (r *deadPoolReaper) stop() {
+	close(r.stopChan)
+	<-r.doneStoppingChan
+}
+
+func (r *deadPoolReaper) loop() {
+	// Reap
+	if err := r.reap(); err != nil {
+		logError("dead_pool_reaper.reap", err)
+	}
+
+	// Begin reaping periodically
+	ticker := time.NewTicker(ReapPeriod)
+
+	for {
+		select {
+		case <-r.stopChan:
+			close(r.doneStoppingChan)
+			return
+		case <-ticker.C:
+			// Jitter
+			time.Sleep(time.Duration(rand.Intn(30)) * time.Second)
```

Comment:
This method of jitter isn't ideal b/c you can't stop while it's sleeping. This would manifest as a hung processes when you issue a sigterm command. recommend adding jitter to ticker.